### PR TITLE
Fix API for the zRAM module

### PR DESCRIPTION
**Pull request by vojtechtrefny**
zRAM module API updated to be consistent with rest of UDisks

### CODE REVIEW

File: modules/zram/data/org.freedesktop.UDisks2.zram.xml [hunk 1]
```diff
@@ -137,24 +137,23 @@
       Values set during device initialisation
```
Comment (vpodzime) [inline modules/zram/data/org.freedesktop.UDisks2.zram.xml]: **to** CamelCase

File: modules/zram/data/org.freedesktop.UDisks2.zram.xml [hunk 2]
```diff
@@ -41,6 +41,7 @@
         @sizes: Requested sizes (in bytes) for created zRAM devices.
```
Comment (vpodzime) [inline modules/zram/data/org.freedesktop.UDisks2.zram.xml]: Extra or missing ``'`` in the commit message.

Comment (vpodzime) [inline modules/zram/data/org.freedesktop.UDisks2.zram.xml]: return**s** nothing

File: modules/zram/udiskslinuxmanagerzram.c [hunk 3]
```diff
@@ -310,6 +310,56 @@ delete_conf_files (GError **error)
...
+  gchar **zram_p = NULL;
+  gint next_obj = 0;
+  gboolean success = TRUE;
+  gint num_zrams = 0;
+  UDisksObject **objects = NULL;
+
+  num_zrams = g_strv_length ((gchar **) user_data);
+  objects = g_new0 (UDisksObject *, num_zrams);
+
+  for (zram_p = (gchar **) user_data; *zram_p != NULL; zram_p++, next_obj++)
```
Comment (vpodzime) [inline modules/zram/udiskslinuxmanagerzram.c]: Would be nice to either describe what ``user_data`` contains here or store it in a properly named local variable of type ``gchar **``.

File: src/udisksdaemon.c [hunk 4]
```diff
@@ -1329,6 +1328,44 @@ udisks_daemon_wait_for_object_sync (UDisksDaemon          *daemon,
...
+{
+  UDisksObject *ret;
+
+  ret = wait_for_objects (daemon,
+                          (UDisksDaemonWaitFuncGeneric) wait_func,
+                          user_data,
+                          user_data_free_func,
+                          timeout_seconds,
+                          error);
+  return (UDisksObject *) ret;
```
Comment (vpodzime) [inline src/udisksdaemon.c]: The type cast should be either dropped or moved above when the value is stored in the variable. Or just use ``return (UDisksObject *) wait_for_objects (...)``

File: src/udisksdaemon.c [hunk 5]
```diff
@@ -1329,6 +1328,44 @@ udisks_daemon_wait_for_object_sync (UDisksDaemon          *daemon,
...
+{
+  UDisksObject **ret = NULL;
+
+  ret = wait_for_objects (daemon,
+                          (UDisksDaemonWaitFuncGeneric) wait_func,
+                          user_data,
+                          user_data_free_func,
+                          timeout_seconds,
+                          error);
+  return (UDisksObject **) ret;
```
Comment (vpodzime) [inline src/udisksdaemon.c]: Same here.

File: modules/zram/data/org.freedesktop.UDisks2.zram.xml (hunk 2 above)
Comment (vojtechtrefny) [inline modules/zram/data/org.freedesktop.UDisks2.zram.xml]: Fixed, thanks.

File: modules/zram/data/org.freedesktop.UDisks2.zram.xml (hunk 1 above)
Comment (vojtechtrefny) [inline modules/zram/data/org.freedesktop.UDisks2.zram.xml]: Fixed.

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Pull request merged**